fedimint-api/src/config.rs

```diff
@@ -326,21 +477,48 @@ where
             let (dkg, step) = Dkg::new(group, our_id, peers, *threshold, rng);
             if let DkgStep::Messages(messages) = step {
                 for (peer, msg) in messages {
-                    connections.send(&[peer], (key.clone(), msg)).await?;
+                    connections
+                        .send(
+                            &[peer],
+                            ConfigGenPeerMsg::DistributedGen((
+                                serde_json::to_string(key).expect("serialization can't fail"),
+                                msg.to_msg(),
+                            )),
+                        )
+                        .await?;
                 }
             }
             dkgs.insert(key.clone(), dkg);
         }
 
         // process steps for each key
+        // TODO: fix error handling here; what do we do on a malfunctining peer when building the federation?
```

Comment:
Right now panicking is fine, but I would also include the misbehaving `PeerId`.  We need to eventually build better handling to eliminate the misbehaving peer from DKG.